subscription_oca/models/sale_subscription.py

```diff
@@ -181,7 +181,7 @@ def _compute_name(self):
             slash = "/" if template_code and code else ""
             record.name = "{}{}{}".format(template_code, slash, code)
 
-    @api.depends("template_id", "date_start")
+    @api.depends("template_id.recurring_rule_boundary", "date_start")
```

Comment:
@PicchiSeba Not sure about this as a change on the template value will change all subscriptions (is it really wanted ?).

Reply:
@rousseldenis 

I made that change because I found myself in the following scenario:

1. Create a subscription
2. Assign a subscription template having duration set to "Forever"
3. Go back to the subscription and see that the "Finish date" field hasn't been set
4. You realize you made a mistake and the template actually has an end date. So you go to the template and set the duration to "Fixed"  with a rule count of "6 months"
5. Returning to the subscription still shows no end date

I find this to be confusing from a user perspective. I realize this could create issues for Subscriptions having the state "in progress".

We could block writes to a certain template if it is assigned to an active subscription.

I'm really interested in your feedback, and sorry for taking this long to answer.

Reply:
Or block writes at all if it's assigned to at least a subscription